core/model/src/main/java/org/eclipse/sw360/antenna/api/service/ServiceFactory.java

```diff
@@ -0,0 +1,128 @@
+    private final ConcurrentMap<ProxySettings, HttpClient> httpClients;
+
+    /**
+     * Creates a new instance of {@code ServiceFactory} with default settings.
+     */
+    public ServiceFactory() {
+        this(new HttpClientFactoryImpl());
+    }
+
+    /**
+     * Creates a new instance of {@code ServiceFactory} that uses the provided
+     * factory to create new HTTP clients.
+     *
+     * @param httpClientFactory the HTTP client factory
+     */
+    ServiceFactory(HttpClientFactory httpClientFactory) {
+        this.httpClientFactory = httpClientFactory;
+        httpClients = new ConcurrentHashMap<>();
+    }
+
+    /**
+     * Returns an initialized JSON object mapper. This mapper can (and should)
+     * be used by all components having to deal with JSON serialization.
+     *
+     * @return the JSON object mapper
+     */
+    public static ObjectMapper getObjectMapper() {
+        return LazyMapperHolder.MAPPER;
+    }
+
+    /**
+     * Returns an initialized HTTP client that supports the given proxy
+     * configuration. This method returns the same client instance when asked
+     * for the same proxy configuration.
+     *
+     * @param useProxy  flag whether a proxy should be used
+     * @param proxyHost the proxy host
+     * @param proxyPort the proxy port
+     * @return the {@code HttpClient} supporting this proxy configuration
+     */
+    public HttpClient createHttpClient(boolean useProxy, String proxyHost, int proxyPort) {
+        ProxySettings proxySettings = ProxySettings.fromConfig(useProxy, proxyHost, proxyPort);
+        return httpClients.computeIfAbsent(proxySettings, this::createHttpClient);
+    }
+
+    /**
+     * Creates a new HTTP client that is configured with the proxy settings
+     * passed in.
+     *
+     * @param settings the proxy settings
+     * @return the new {@code HttpClient}
+     */
+    private HttpClient createHttpClient(ProxySettings settings) {
+        HttpClientConfig clientConfig = createHttpClientConfig(settings);
+        return httpClientFactory.newHttpClient(clientConfig);
+    }
+
+    /**
+     * Creates a configuration for a new HTTP client based on the given proxy
+     * settings. As the JSON mapper needed by the client the one managed by
+     * this factory is used.
+     *
+     * @param settings the proxy settings
+     * @return the {@code HttpClientConfig} for these proxy settings
+     */
+    private static HttpClientConfig createHttpClientConfig(ProxySettings settings) {
+        return HttpClientConfig.basicConfig()
+                .withObjectMapper(getObjectMapper())
+                .withProxySettings(settings);
+    }
+
+    /**
+     * Application of the Initialization-on-demand holder idiom for the JSON
+     * object mapper.
+     */
+    private static class LazyMapperHolder {
+        private static final ObjectMapper MAPPER = new ObjectMapper()
+                .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    }
```

Comment:
I think this class is superfluous and only used once (if I haven't missed an occurrence) and could just be in the "getObjectMapper()" method.

Reply:
This is tricky, but the class plays indeed an important role for the safe initialization of the ObjectMapper. This is the so-called [Initialization-on-demand holder idiom](https://de.wikipedia.org/wiki/Initialization-on-demand_holder_idiom)